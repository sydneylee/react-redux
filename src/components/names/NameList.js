import React, {Component} from 'react';
import PropTypes from 'prop-types';


class NameList extends Component {
    static defaultProps = {
        names :['aaa','bbb','ccc']
    };

    static propTypes = {};

    renderList=()=>{
       const names = this.props.names;
       return names.map((name, i)=>{
           return (<li key={i} >{name}</li>);
       })

    };

    state = {};

    render() {
        //const {names} = this.props;
        return (
            <ul>
                {this.renderList()}
            </ul>
        );
    }
}

export default NameList;
