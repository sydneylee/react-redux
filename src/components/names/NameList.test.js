import React from 'react';
import renderer from 'react-test-renderer';
import  NameList from './NameList';

describe('NameList', () => {
    let component = null;

    it('renders correctly', () => {
        component = renderer.create(<NameList names={["벨로퍼트", "김민준"]} />);
    });

    it('matches snapshot', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
