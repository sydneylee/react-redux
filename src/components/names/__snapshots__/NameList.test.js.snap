// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameList matches snapshot 1`] = `
<ul>
  <li>
    undefined  undefined
  </li>
  <li>
    undefined  undefined
  </li>
</ul>
`;
